staging/src/k8s.io/component-helpers/scheduling/corev1/nodeaffinity/nodeaffinity_test.go

```diff
@@ -218,8 +218,8 @@ func TestPreferredSchedulingTermsScore(t *testing.T) {
 				},
 				&field.Error{
 					Type:   field.ErrorTypeInvalid,
-					Field:  "[2].matchExpressions[0]",
-					Detail: `invalid label key "invalid key": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')`,
+					Field:  "[2].matchExpressions[0].key",
+					Detail: `name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')`,
```

Comment:
"name part" what other part of the key is there?

Reply:
The "name part" generated by `IsQualifiedName`, here is the whole key (`.key` indicates it).
https://github.com/kubernetes/kubernetes/blob/65395137b5c4354893408dcc8d078954210ffeff/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L67-L69
https://github.com/kubernetes/kubernetes/blob/a1f8dc41eff5cb1e6c1e6e23b093dcf75dd057cd/staging/src/k8s.io/apimachinery/pkg/labels/selector.go#L874-L876

Reply:
Will a user understand this from the message?

Reply:
can we just remove "name part"?